emmet-core/emmet/core/mpid.py

```diff
@@ -41,7 +41,7 @@ def __eq__(self, other: object):
             return self.parts == MPID(other).parts
 
     def __str__(self):
-        return "-".join(self.parts)
+        return "-".join((str(i) for i in self.parts))
```

Comment:
Can you add a test that will fail without this bug fix?

Reply:
This was failing because `join()` would not work with the second entry in `parts` being an integer. Added a test to check string conversion.